pkg/internal/codegen/parse/util.go

```diff
@@ -395,3 +425,45 @@ func parseScaleParams(t *types.Type) (map[string]string, error) {
 	}
 	return nil, fmt.Errorf(jsonPathError)
 }
+
+// ready subresource requires name,type,JSONPath fields and rest of the field are optional
+// +kubebuilder:printcolumn:ready:name=<name>,type=<type>,description=<desc>,JSONPath:<.spec.Name>,priority=<0|1>,format=<format>
+func parsePrintColumnParams(t *types.Type) v1beta1.CustomResourceColumnDefinition {
+	config := v1beta1.CustomResourceColumnDefinition{}
+	for _, comment := range t.CommentLines {
+		if strings.Contains(comment, "+kubebuilder:printcolumn:ready") {
+			parts := strings.Replace(comment, "+kubebuilder:printcolumn:ready:", "", -1)
+			part := strings.Split(parts, ",")
+			if len(part) < 3 {
+				fmt.Println(fmt.Errorf("Expected required fields in this context: %s", comment))
+			}
+			for _, s := range part {
+				arr := strings.Split(s, "=")
+				switch arr[0] {
+				case "printColumnName":
+					config.Name = arr[1]
+				case "printColumnType":
+					config.Type = arr[1]
+				case "printColumnFormat":
+					config.Format = arr[1]
```

Comment:
There are some valid allowed values specified in the [spec](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#additional-printer-columns). We should validate against those.